src/jit/hwintrinsicxarch.cpp

```diff
@@ -754,6 +760,70 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
             }
             break;
 
+        case NI_SSE2_ConvertScalarToVector128Double:
+        {
+            assert(sig->numArgs == 2);
+            assert(getBaseTypeOfSIMDType(sig->retTypeSigClass) == TYP_DOUBLE);
+
+            argList = info.compCompHnd->getArgNext(sig->args);
+            CorInfoType corType =
+                strip(info.compCompHnd->getArgType(sig, argList, &argClass)); // type of the second argument
+
+            baseType = JITtype2varType(corType);
+
+#ifdef _TARGET_X86_
+            if (varTypeIsLong(JITtype2varType(corType)))
+            {
+                return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
+            }
+#endif // _TARGET_X86_
+
+            if (baseType == TYP_STRUCT)
+            {
+                baseType = getBaseTypeOfSIMDType(argClass);
```

Comment:
I believe that the baseType is always `double` on `Sse2.ConvertScalarToVector128Double`.

Reply:
Yes, that's the case however I try to write code which will be easy to refactor or abstract to handle more cases later see issue #16014. If it is acceptable I would prefer to defer to refactoring PR

Reply:
Never mind. `CVTSS2SD xmm, xmm/m32` and `CVTSI2SD xmm, reg/m64` need different encoding.

Reply:
In this case I use argType as baseType to use it later during instruction emission